src/Core/System/Consent/Service/ConsentService.php

```diff
@@ -0,0 +1,211 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\Consent\Service;
+
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\System\Consent\ConsentContext;
+use Shopware\Core\System\Consent\ConsentDefinition;
+use Shopware\Core\System\Consent\ConsentException;
+use Shopware\Core\System\Consent\ConsentRepository;
+use Shopware\Core\System\Consent\ConsentScope;
+use Shopware\Core\System\Consent\ConsentStatus;
+use Shopware\Core\System\Consent\DTO\ConsentState;
+use Shopware\Core\System\Consent\DTO\ConsentStateLogRecord;
+use Shopware\Core\System\Consent\Event\ConsentAcceptedEvent;
+use Shopware\Core\System\Consent\Event\ConsentRevokedEvent;
+
+/**
+ * @internal
+ */
+#[Package('data-services')]
+class ConsentService
+{
+    /**
+     * @var array<string, ConsentDefinition>
+     */
+    private array $consentDefinitions;
+
+    /**
+     * @var array<string, ConsentState>
+     */
+    private ?array $states = null;
+
+    /**
+     * @param iterable<ConsentDefinition> $consentDefinitions
+     */
+    public function __construct(
+        iterable $consentDefinitions,
+        private readonly ConsentRepository $consentRepository,
+        private readonly EventDispatcherInterface $eventDispatcher
+    ) {
+        $definitions = [];
+        foreach ($consentDefinitions as $definition) {
+            $definitions[$definition->getName()] = $definition;
+        }
+        $this->consentDefinitions = $definitions;
+    }
+
+    /**
+     * @return array<ConsentState>
+     */
+    public function list(ConsentContext $context): array
+    {
+        $states = $this->fetchStates();
+
+        return array_map(function (ConsentDefinition $consent) use ($context, $states) {
+            $identifier = $context->getIdentifierForScope($consent->getScope());
+            $key = $this->key(
+                $consent->getName(),
+                $consent->getScope()->value,
+                $identifier
+            );
+
+            return $states[$key] ?? new ConsentState(
+                name: $consent->getName(),
+                scope: $consent->getScope(),
+                identifier: $identifier,
+                status: ConsentStatus::REQUESTED,
+                actorId: null,
+            );
+        }, $this->consentDefinitions);
+    }
+
+    public function getConsentState(string $name, ?string $identifier = null): ConsentState
+    {
+        $consent = $this->getConsentDefinition($name);
+
+        if ($consent->getScope() !== ConsentScope::GLOBAL && $identifier === null) {
+            throw ConsentException::identifierRequired();
+        }
+
+        $states = $this->fetchStates();
+
+        $key = $this->key($consent->getName(), $consent->getScope()->value, $identifier);
+
+        if (isset($states[$key])) {
+            return $states[$key];
+        }
+
+        return new ConsentState(
```

Comment:
i assume we do not store requested consents in the DB so it easy to add new consents and we do not have to write a DB migration to add that? We should document that here somewhere in this class, that totally makes sense, but is definetly not obvious